### PR TITLE
Make `ReportButton` a CTA button below the list

### PR DESCRIPTION
**Pull request by kalwabed**
Closes https://github.com/kawalcovid19/wargabantuwarga.com/issues/155

## Current Tasks

- [x] Add icon
- [x] Unify all buttons into a single CTA button below the list
- [x] Change Goggle Form query based on categories

### CODE REVIEW

File: components/contact-details.tsx [hunk 1]
```diff
@@ -38,9 +39,10 @@ const ReportButton = (props: ReportButtonProps) => (
     <button
       className="bg-white rounded-md font-medium text-indigo-600 hover:text-indigo-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
       onClick={props.onClick}
+      title="Laporkan kesalahan"
```
Comment (resir014) [inline components/contact-details.tsx:42]:
```suggestion
      aria-label="Laporkan kesalahan"
```

I think `aria-label` is a slightly better practice over `title`.

**Review by resir014 — DISMISSED** [review]
Only one minor thing from me. The design of this button will also need to be paired with the copy buttons in #162 once that PR gets merged.

**Review by zainfathoni — CHANGES_REQUESTED** [review]
@kalwabed, after my call recently with the team, we decided to put the action as a CTA down below the list since the actions for every button are the same.

Could you please update your implementation to comply with these new criteria?

File: components/ReportButton.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        type="button"
+      >
+        <ExclamationCircleIcon aria-hidden="true" className="w-5 h-5" />{" "}
+        <span>Laportkan kesalahan</span>
+      </button>
+    </div>
+  );
+};
+
+export default ReportButton;
```
Comment (zainfathoni) [inline components/ReportButton.tsx:40]:
```suggestion
```

Please use named export to guarantee to autocomplete in VS Code editor. 😁

File: components/ReportButton.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Contact } from "../lib/provinces";
+
+import { ExclamationCircleIcon } from "@heroicons/react/outline";
+
+type ReportButtonProps = {
+  contact: Contact;
+  provinceName: string;
+};
+
+const ReportButton = (props: ReportButtonProps) => {
```
Comment (zainfathoni) [inline components/ReportButton.tsx:10]:
```suggestion
export const ReportButton = (props: ReportButtonProps) => {
```

Please use named export to guarantee to autocomplete in VS Code editor. 😁

File: pages/provinces/[provinceSlug]/[contactSlug].tsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { BackButton } from "../../../components/layout/back-button";
 import { Page } from "../../../components/layout/page";
 import { PageContent } from "../../../components/layout/page-content";
 import { PageHeader } from "../../../components/layout/page-header";
+import ReportButton from "../../../components/ReportButton";
```
Comment (zainfathoni) [inline pages/provinces/[provinceSlug]/[contactSlug].tsx:7]:
```suggestion
import ReportButton from "../../../components/report-button";
```

Please rename the file into `report-button.tsx` to comply with our de facto file naming convention.

File: components/contact-details.tsx [hunk 5]
```diff
@@ -69,65 +33,31 @@ const DescriptionItem = (props: DescriptionItemProps) => {
...
-        <DescriptionItem
-          label="Tautan"
-          onClick={_reportButtonHandler}
-          value={contact.link}
-        />
+        <DescriptionItem label="Penyedia" value={contact.penyedia} />
+        <DescriptionItem label="Keterangan" value={contact.keterangan} />
+        <DescriptionItem label="Lokasi" value={contact.lokasi} />
+        <DescriptionItem label="Kontak" value={contact.kontak} />
+        <DescriptionItem label="Alamat" value={contact.alamat} />
```
Comment (zainfathoni) [inline components/contact-details.tsx:49]:
```suggestion
        <DescriptionItem label="Kontak" value={contact.kontak} withCopyButton />
        <DescriptionItem label="Alamat" value={contact.alamat} withCopyButton />
```

Please keep the `withCopyButton` props intact. 😅

File: components/ReportButton.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <div className="flex justify-center">
+      <button
+        aria-label="Laporkan kesalahan"
+        className="relative z-10 px-2 h-8 mt-2 bg-gray-100 text-gray-600 hover:bg-blue-600 hover:text-white flex items-center justify-center rounded text-xs space-x-1"
+        onClick={reportButtonHandler}
+        type="button"
+      >
+        <ExclamationCircleIcon aria-hidden="true" className="w-5 h-5" />{" "}
+        <span>Laportkan kesalahan</span>
```
Comment (zainfathoni) [inline components/ReportButton.tsx:34]:
```suggestion
        <span>Laporkan kesalahan</span>
```

**Review by zainfathoni — CHANGES_REQUESTED** [review]
I have some minor but essentials changes here. 😁 

Thanks, anyway, @kalwabed!

**Review by zainfathoni — CHANGES_REQUESTED** [review]
I also found out that somehow the `text-indigo-600 hover:text-indigo-500` styling in the `Kontak` and `Tautan` are missing, but I didn't know which parts of the code that changes this. 🤔 

Screenshot 2021-07-19 at 23 31 01

Could it possibly because this PR is not updated from the latest `main` branch?

File: components/contact-details.tsx (hunk 5 above)
Comment (kalwabed) [reply components/contact-details.tsx:49]: Oops, it seems I forgot to add it. I will change soon. 😅

**Comment by kalwabed** [comment]
> I also found out that somehow the `text-indigo-600 hover:text-indigo-500` styling in the `Kontak` and `Tautan` are missing, but I didn't know which parts of the code that changes this. 
> 
> Screenshot 2021-07-19 at 23 31 01
> 
> Could it possibly because this PR is not updated from the latest `main` branch?

I think it's because we are using the `DescriptionItem` component, instead of using the `DescriptionLink` component? I found it in [`65757b5` (#75)](https://github.com/kawalcovid19/wargabantuwarga.com/pull/75/commits/65757b54ca0592ac787d4ebae69dc9a68e121066#diff-6d6ddc00d0535a4ddc8fffadbf7ddf417ea42a5415c2ce12802c0201d510dcbbR39).

**Comment by zainfathoni** [comment]
> After I inspect elements I don't find Kontak or Tautan contains <a> tags, maybe this is the problem?

No, it is definitely an `<a>` tag.
Screenshot 2021-07-20 at 10 40 29

**Comment by zainfathoni** [comment]
There must be something in your PR that breaks that styling because in my [deployment preview](https://deploy-preview-191--wargabantuwarga.netlify.app/provinces/dkijakarta-1/3rmedika-gas-130) the styling works as expected.
Try comparing it to your [deployment preview](https://deploy-preview-171--wargabantuwarga.netlify.app/provinces/dkijakarta-1/3rmedika-gas-130).
I will also try investigating this issue from my side.

**Comment by mazipan** [comment]
`Laporkan kesalahan` somehow not well explained in my opinion.

As a user, I don't know what 'kesalahan' do you mean.
I need more context, what type of 'kesalahan' I can report.
What the expectation of the report process.

File: components/contact-details.tsx [hunk 7]
```diff
@@ -11,6 +11,20 @@ type ContactDetailsProps = {
   provinceName: string;
 };
 
+// ? please don't remove this ReportButton until issue is resolved. - @kalwabed
+// reference: https://github.com/kawalcovid19/wargabantuwarga.com/pull/171
+// const ReportButton = (props: ReportButtonProps) => (
```
Comment (kalwabed) [inline components/contact-details.tsx:16]: If we remove this, the style will break. Mas @zainfathoni, maybe this is an issue from the JIT Engine? I don't know. 😅

Comment (zainfathoni) [reply components/contact-details.tsx:16]: Probably 😅

Let's keep it for now and try to resolve it separately. 😁

**Comment by zainfathoni** [comment]
> `Laporkan kesalahan` somehow not well explained in my opinion.
> 
> 
> 
> As a user, I don't know what 'kesalahan' do you mean.
> 
> I need more context, what type of 'kesalahan' that I can report.
> 
> What the expectation of the report process.
> 

Since this button links to this form:

image

Shall we rename it into **Adukan Informasi**?

Do you think it's clearer or not Mas @mazipan ?

> Also about the visibility, since it put on the bottom, for a mobile device user, it might be hard to see the button

Shall we put it at tho top-right corner beside the sorting dropdown, Mas @mazipan @kalwabed ?

**Comment by mazipan** [comment]
If we want to add a phrase before the action, I think it should be like:

"Menemukan kesalahan informasi? silahkan adukan kesalahan"

But yes, it's NOT as simple as a single button 😂

**Pull request merged**